core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4Connection.java

```diff
@@ -1876,9 +1876,57 @@ public SQLXML createSQLXML() throws SQLException {
 		return null;
 	}
 
-	public boolean isValid(int timeout) throws SQLException {
-		// TODO Auto-generated method stub
-		return false;
+    public boolean isValid(int timeout) throws SQLException {
+        if (props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(props_);
+            props_.t4Logger_.logp(Level.FINE, "TrafT4Connection", "isValid", "iValid", p);
+        }
+        if (props_.getLogWriter() != null) {
```

Comment:
no,  the log will only be written when user of client set the log level to true and log writer is not null,